chain/src/txhashset/txhashset.rs

```diff
@@ -973,7 +966,7 @@ impl<'a> HeaderExtension<'a> {
 			self.head.height,
 		);
 
-		let header_pos = pmmr::insertion_to_pmmr_index(header.height + 1);
+		let header_pos = 1 + pmmr::insertion_to_pmmr_index(header.height);
```

Comment:
huh? we avoid the original + 1 because the argument is now 0-based instead of 1-based and we add 1 because rewind still expects a 1-based position. there is no "next pos" involved here; the original header.height + 1 was needed because the first header MMR entry is genesis which has header.height 0 which was at 1-based position 1.